sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -234,7 +237,7 @@ case class InsertIntoHadoopFsRelationCommand(
     if (fs.exists(staticPrefixPath) && !committer.deleteWithJob(fs, staticPrefixPath, true)) {
       throw QueryExecutionErrors.cannotClearOutputDirectoryError(staticPrefixPath)
     }
-    // now clear all custom partition locations (e.g. /custom/dir/where/foo=2/bar=4)
+    // now clear custom static partition locations (e.g. /custom/dir/where/foo=1)
```

Comment:
This is changed because after this PR, only overwriting the static partition will run into deleteMatchingPartitions.